lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -651,6 +651,24 @@ function dosomething_global_url($path, $options = NULL) {
 }
 
 /**
+ * Build a global node URL that adds the prefix and a fragment
+ *
+ * @param string $nid
+ *    The nid of the node
+ * @param string $language
+ *    The language of the URL
+ * @param string $fragment
+ *    Optional fragment to append
+ */
+function dosomething_global_node_url($nid, $language, $fragment = NULL) {
+  $raw_url = 'node/' . $nid;
+  if (isset($fragment)) {
+    $raw_url .= '#' . $fragment;
```

Comment:
- I've caught it not properly apply the path alias on nodes (on global / translated titles)
- It often makes duplicate prefixes (like what the referenced bug here is describing)
- I've also had problems with it applying the correct language to a URL

This is why for many days now I've been manually building the URL with path_alias and get prefix, because it _always_ works fine without problem. Whereas url() consistently gives me problems and does not behave well with global functionality.